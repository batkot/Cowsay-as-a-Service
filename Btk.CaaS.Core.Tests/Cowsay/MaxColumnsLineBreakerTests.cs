using ApprovalTests;
using ApprovalTests.Reporters;
using Btk.CaaS.Core.Cowsay;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Xunit;

namespace Btk.CaaS.Core.Tests.Cowsay
{
    [UseReporter(typeof(DiffReporter))]
    public class MaxColumnsLineBreakerTests
    {
        [Fact]
        public void message_shorter_than_threshold_should_not_split()
        {
            var lineBreaker = new MaxColumnsLineBreaker(5);
            var msg = "123";

            var result = lineBreaker.Break(msg);
            Assert.Equal(1, result.Count());
            Assert.Equal("123", result.First());
        }

        [Fact]
        public void message_longer_than_threshold_should_split()
        {
            var lineBreaker = new MaxColumnsLineBreaker(5);
            var msg = "12345678901";

            var result = lineBreaker.Break(msg);
            Assert.Equal(3, result.Count());
            Approvals.VerifyAll(result, "label");
        }

    }
}
