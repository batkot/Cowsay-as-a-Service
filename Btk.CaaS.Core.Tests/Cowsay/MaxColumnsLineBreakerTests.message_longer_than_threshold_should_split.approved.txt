label[0] = 12345
label[1] = 67890
label[2] = 1
